hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3074,6 +3054,17 @@ func (r *HostedClusterReconciler) delete(ctx context.Context, hc *hyperv1.Hosted
 		}
 	}
 
+	// Cleanup Platform specifics.
+	p, err := platform.GetPlatform(hc, r.AvailabilityProberImage, r.TokenMinterImage)
```

Comment:
in our deletion model the `controlPlaneNamespace` is wiped out, so there's no need to target particular resources for deletion. I see why this is needed here for the AgentNamespace, which makes me keep questioning if the Agent clusterRole and ClusterRoleBinding should be owned by Hypershift vs an out of band requirement. No a blocker to me anyways.

Reply:
https://github.com/openshift/hypershift/pull/1137#discussion_r825463831

Reply:
ok, I meant any clusterRole/Role that need that handles perms in the agentNamespace. But anyways It makes things easier and we can always revisit later if necessary.